### PR TITLE
SYCL Warnings

### PR DESCRIPTION
**Pull request by rchen20**
# Summary

- This PR is a refactoring?
- It does the following:
  - Updates to the SYCL 2020 API. Memory accessors have changed and were generating deprecation warnings.
  - Requires similar CAMP update https://github.com/LLNL/camp/pull/147.

### CODE REVIEW

File: include/RAJA/policy/sycl/launch.hpp
Comment (rchen20) [inline include/RAJA/policy/sycl/launch.hpp:177]: Hi @homerdin, I was trying to eliminate SYCL warnings from RAJA (and CAMP https://github.com/LLNL/camp/pull/147), and had to update some of our SYCL functionality to use the 2020 API. Most of the changes were straightforward, except this new `get_multi_ptr`. Specifically, I couldn't find documentation on what `access::decorated` type to pass in; `yes` seems to work, but it would be good to know what exactly this is doing and whether this actually gets us the shared memory we are expecting from this call. Please let me know if this is ok, and feel free to change this to something better.

Comment (long58) [reply include/RAJA/policy/sycl/launch.hpp:177]: From the [sycl 2020 specs](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#command): 

> The multi_ptr class exposes 3 flavors of the same interface. If the value of access::decorated is access::decorated::no, the interface exposes pointers and references type that are not decorated by an address space. If the value of access::decorated is access::decorated::yes, the interface exposes pointers and references type that are decorated by an address space. The decoration is implementation dependent and relies on device compiler extensions. The decorated type may be distinct from the non-decorated one. For interoperability with the [SYCL backend](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#backend), users should rely on types exposed by the decorated version. If the value of access::decorated is access::decorated::legacy, the 1.2.1 interface is exposed. This interface is deprecated.

Comment (rchen20) [reply include/RAJA/policy/sycl/launch.hpp:177]: Thanks @long58, that's what I was seeing as well, and was hoping for a better explanation of `yes`, `no`, and `legacy`.

Comment (rhornung67) [reply include/RAJA/policy/sycl/launch.hpp:177]: FYI, I just tried this branch and the corresponding camp branch in a RAJAPerf build on corona and it built and ran correctly. I'll have a RAJAPerf branch ready to merge after the RAJA and camp PRs are merged.

"Three flavors"....who writes their documentation?!? Although the code changes to taste a bit like "yes" 😄

**Comment by rhornung67** [comment]
@rchen20 before merging this, please pull BLT v0.6.2 into RAJA also. Then, tag a v2024.02.1 release.

File: include/RAJA/policy/sycl/launch.hpp
Comment (rhornung67) [reply include/RAJA/policy/sycl/launch.hpp:177]: @long58 thanks for citing the SYCL spec reference.

**Comment by rhornung67** [comment]
@rchen20   Feel free to pull the new camp in here after it is released if you want to limit the number of PRs to merge.

File: blt
Comment (artv3) [inline blt:1]: Is blt supposed to be updated with this PR?

**Comment by rchen20** [comment]
Hey @homerdin, when you get a chance would you mind taking a quick look at these SYCL 2020 API updates? Without these changes, we get multiple deprecation warnings during compilation. Thanks!

**Review by rhornung67 — APPROVED** [review]

**Review by homerdin — APPROVED** [review]
@rchen20 The changes look go to me.  I've asked around about `access::decorated`, but haven't been able to get any clarification yet.

**Comment by rhornung67** [comment]
@rchen20 i'm prepping the v2024.02.1 RC branch. It will be next in line to merge after this PR is merged.

File: include/RAJA/policy/sycl/launch.hpp
Comment (homerdin) [reply include/RAJA/policy/sycl/launch.hpp:177]: @rchen20 I was told that they are adding the corresponding __local, __private, __global and __constant to pointer/reference declarations.  `yes` should be correct.

Comment (rchen20) [reply include/RAJA/policy/sycl/launch.hpp:177]: > @rchen20 I was told that they are adding the corresponding __local, __private, __global and __constant to pointer/reference declarations. `yes` should be correct.

Thanks! To be clear, do you mean that `get_multi_ptr<::sycl::access::decorated::yes>` returns a pointer or reference decorated with one of the above qualifiers (depending on the base pointer/ref type)?

**Pull request merged**
